functions/query/query.go

```diff
@@ -153,3 +152,72 @@ func toNumber(input string, defaultValue int) (int, error) {
 
 	return strconv.Atoi(input)
 }
+
+func validateRequest(eventBody []byte) (*request, error) {
+
+	requestMap := make(map[string]interface{})
+	if err := json.Unmarshal(eventBody, &requestMap); err != nil {
+		return nil, errors.Wrap(err, "Request body must be a valid JSON object")
+	}
+
+	request := request{}
+	metric, ok := requestMap["metric"]
+	delete(requestMap, "metric")
+	if !ok {
+		return nil, errors.New("Request object is missing 'metric' field")
+	}
+	request.Metric, ok = metric.(string)
+	if !ok {
+		return nil, errors.New("'metric' field must be a string")
+	}
+	var err error
+	request.FilterExpression, err = validateOptionalString(requestMap, "filter_expression")
+	if err != nil {
+		return nil, err
+	}
+	request.Step, err = validateOptionalString(requestMap, "step")
+	if err != nil {
+		return nil, err
+	}
+	request.StartTime, err = validateOptionalString(requestMap, "start_time")
+	if err != nil {
+		return nil, err
+	}
+	request.EndTime, err = validateOptionalString(requestMap, "end_time")
+	if err != nil {
+		return nil, err
+	}
+	request.Last, err = validateOptionalString(requestMap, "last")
```

Comment:
If we are already validating, let's validate the user won't pass both `start & end` and `last`.